presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -391,7 +392,10 @@ protected void setup(Binder binder)
         binder.bind(FinalizerService.class).in(Scopes.SINGLETON);
 
         // Spiller
-        binder.bind(SpillerFactory.class).to(BinarySpillerFactory.class).in(Scopes.SINGLETON);
+        install(conditionalModule(
+                FeaturesConfig.class,
+                config -> BINARY_FILE.equalsIgnoreCase(config.getSpillerImplementation()),
```

Comment:
What happens otherwise, no `SplillerFactory` is povided? Why not to introduce simple true/false switch instead of string matching?